src/uucore/src/lib/features/perms.rs

```diff
@@ -188,6 +189,77 @@ pub struct ChownExecutor {
     pub dereference: bool,
 }
 
+#[cfg(test)]
+pub fn check_root(path: &Path, would_recurse_symlink: bool) -> bool {
+    is_root(path, would_recurse_symlink)
+}
+
+fn is_root(path: &Path, would_recurse_symlink: bool) -> bool {
+    // Prohibit only if:
+    // (--preserve-root and -R present) &&
+    // (path canonicalizes to "/") &&
+    // (
+    //     (path is a symlink && would recurse on this symlink) ||
+    //     (path is not a symlink)
+    // )
+    // The first clause is checked by the caller, the second and third clause is checked here.
+    // The caller has to evaluate -P/-H/-L into 'would_recurse_symlink'.
+    // Recall that canonicalization resolves both relative paths (e.g. "..") and symlinks.
```

Comment:
I think this makes sense as a docstring. We don't really need to mention `--preserve-root` and `-R` here, because we're not in the context of a specific util.

Reply:
Why not mention `--preserve-root` in the function that has the express purpose of the flag? Also, this serves as a reminder of how to use this function correctly.

I'm happy to turn this a docstring.

Reply:
I guess mentioning them is fine but it should then also mention on which utils that flag is used.

Reply:
Good idea, will do!